### DISCUSSION TITLE
Phase banner

### DISCUSSION THREAD

**Discussion by govuk-design-system** [alphagov/govuk-design-system#4973] [Prioritised components and patterns]
Use this issue to discuss [this component in the GOV.UK Design System](https://design-system.service.gov.uk/components/phase-banner/).

**Comment by timpaul** [comment]
There's some evidence here from the GOV.UK Notify team that the beta banner can cause confusion and even put some people off using their service: https://github.com/alphagov/notifications-admin/pull/1838

**Comment by 36degrees** [comment]
Worth noting that they weren't using the beta _banner_, just the beta tag – as per the [product page example](https://github.com/alphagov/product-page-example).

A traditional beta banner might have been fine. It possibly is evidence that that the tag by itself isn't working, though.

**Comment by jbuller** [comment]
It would be useful to have clarity on whether the phase banner (and breadcrumb)s should be inside the main element or not. (I'd say outside)
The Start page pattern on the Design System has it above <main> 
However for example https://www.gov.uk/service-manual/helping-people-to-use-your-service/making-your-service-accessible-an-introduction has it inside the main element, 
This means the Skip to content link only cuts out about half the points of keyboard focus it would otherwise.

**Comment by dashouse** [comment]
@jbuller @stevenaproctor Hi, you'll find this implemented in many different ways across GOV.UK sites as they were all made at different times. If you're using the GOV.UK Design System / GOV.UK Frontend then this page on the design system clarifies the structure of the page we intend people to use  https://design-system.service.gov.uk/styles/layout/#page-wrappers

**Comment by jbuller** [comment]
Thanks for the pointer. 
I'd say it should be clearer on the phase banner page, or link to where it is clarified.
I didn't even find the page you highlighted when I was searching.
Actually, only the 'Customised page template' section of next page has the phase banner html in the larger context, above main element https://design-system.service.gov.uk/styles/page-template/

**Comment by suryanaga** [comment]
So to request a few changes to this:

1. We should clearly state the Alpha phase banner should only be used for prototypes, not working services. Possibly a link to [the service manual's page about the alpha phase](https://www.gov.uk/service-manual/agile-delivery/how-the-alpha-phase-works) would be useful here.
2. In the beta banner, we should recommend linking to both the legacy service and the support channels that are available for users struggling to use the service. (I think @soniaturcotte did something like this recently for a GOV.UK Publisher beta.) Possibly it would be good to design a few options for people to pick from, depending on the setup of their beta.

**Comment by dashouse** [comment]
Thanks @sanjaypoyzer, as this is a direct request for an update to the guidance I've moved this into an issue on the Design System itself so it can be triaged in our next session.

https://github.com/alphagov/govuk-design-system/issues/869

**Comment by drewno-design** [comment]
> @jbuller @stevenaproctor Frontend then this page on the design system clarifies the structure of the page we intend people to use https://design-system.service.gov.uk/styles/layout/#page-wrappers

I am not sure how to interpret the placement of phase banner. Should it be directly underneath header but before main element? 

My main concern is that if the phase banner is in the main element, then it is read by screen reader at the beginning of the main content.

**Comment by jbuller** [comment]
I agree it's still not clear. The phrase "so that they sit directly underneath the header." suggests outside of the `<header>` element and then it says to put it in the same div wrapper as the `<main>` element.
So that means its not supposed to be in any landmark area? 

But yes your reasoning is correct, don't put the phase banner inside the main element to avoid clutter.for assistive technology users on every page.

**Comment by terrysimpson99** [comment]
I propose a rethink of the phase banner. It doesn't satisfy a user need for the services that I'm involved with and doesn't appear to add a business benefit either. I've never understood what it's for and the guidance page doesn't say.

I asked on slack and there were two suggestions:
(1) somebody suggested it would help a user if there is more than one version of the service (such as `www.moneyclaim.gov.uk`
and https://www.gov.uk/make-money-claim) or one service available in two phases. 

(2) somebody suggested it helps if a service is undergoing rapid change.

Scenario 1 makes sense. I can see why scenario 2 exists but for most services and most users it's a moot point. As long as the user can get from the start to the end of the journey there's no benefit in the banner. 

The phase banner contains a feedback link. Here are my thoughts on that: (a) if it's important then there would be a feedback link for live services; (b) on our service in the last month, we got ~11700 instances of feedback, only 26 of those via the feedback link in the phase banner, the rest were from the post-submit feedback; (c) Of the 26 instances of the feedback some of them were requests for assistance and so the feedback link is drawing users away from our 'Get help with this page' link on the same page; (d) having a feedback link in a feedback page is too weird for me but that's what we do.

In summary, I'm raising two separate issues: the first relates to the phase banner; the second relates to in-page feedback. These two issues can be considered separately. What do others think?

**Comment by timpaul** [comment]
Thanks fo the comments @terrysimpson99 - I've fed them back to our Standards and Assurance team here at GDS.

**Comment by terrysimpson99** [comment]
Just an update on this. Some services modify the banner design. This can involve splitting the tag from the feedback link. It can also involve putting the banner below the page content. The hmrc app goes even further and has no phase banner which is what I think we should do. @timpaul has the Standards and Assurance team responded to you?

**Comment by terrysimpson99** [comment]
Somebody previously suggested a phase banner might be useful to users if there's more than one version. However, the current pattern makes no reference to this or how to switch between versions. The Coronavirus dashboard is available in two versions and allows users to switch between them. Unfortunately it's inconsistent because it used an alert for the live site (because there's no phase banner for live). I think this is another reason the phase banner needs a rethink. See: 
![image](https://user-images.githubusercontent.com/48089049/86887940-47a70180-c0f1-11ea-837f-e655eb3de521.png) 
![image](https://user-images.githubusercontent.com/48089049/86888162-a40a2100-c0f1-11ea-8a50-f676bf11b900.png)

**Comment by timpaul** [comment]
Thanks again Terry. I've fed this back to the relevant team in GDS again. I think you're right that beta banners are trying to do multiple things, and not all well. Changing them requires close co-ordination with the Standards and Assessments team here in GDS, so I'll try and arrange a chat with them.

**Comment by terrysimpson99** [comment]
There is inconsistency in how services apply the phase banner guidance. It's common for the feedback link to open in a new window. Some services warn the user by making the link text `feedback (opens in a new window)` others do not warn the user and the link text is merely `feedback`. I enclose an image of the Money Claims service which has a warning and an image of Government Gateway which has no warning.
![image](https://user-images.githubusercontent.com/48089049/94545759-e2bcfc80-0244-11eb-9d14-08e9f4df38cd.png)
![image](https://user-images.githubusercontent.com/48089049/94545794-ed779180-0244-11eb-847f-9b4d1ee16312.png)

I found guidance on the phrase '(opens in a new window)' at: `https://design-system.service.gov.uk/styles/typography/` and `https://design.tax.service.gov.uk/hmrc-design-patterns/open-links-in-a-new-window-or-tab/`. I think it might be more error resistant to update the phase banner example to show `feedback (opens in a new window)`. What do others think?

**Comment by joelanman** [comment]
In the Government Gateway example above, it does open a new tab. Following guidance, it should have the warning text.

I think the question is should the default example of the banner in the Design System open the link in a new tab. Possibly, as users might lose data and their place in the service if it does not.

**Comment by iancuddy** [comment]
I propose some changes to the example text in the Design System phase banner pattern. 

There are currently some issues with the existing content:

- The link 'feedback' doesn't follow the best practice that links should make sense when read out-of-context 
- It's not obvious that clicking 'feedback' would direct the user to a feedback form
- Having the link mid-sentence is not ideal and should be at the end of the sentence to avoid interrupting the user's flow
- The term 'new' is time-sensitive - some services in extended betas have carry this text for a considerable time
- As mentioned above, the link text should state (opens in a new tab) for accessibility reasons
- Do users understand what 'Alpha' or'Beta' means?
- Could the content be made more engaging to encourage users to give their feedback?

A potential improvement could be to draw upon the example text in the Confirmation page pattern where we ask for feedback, which is 'What do you think of this service? (takes 30 seconds)'.  For example, 'Tell us what you think of this service (opens in new tab)'.

**Comment by timpaul** [comment]
Thanks for the feedback @iancuddy. 

The links in the examples in the Design System don't currently open in a new tab, but many services do implement the component that way, so perhaps the examples should reflect that. 

We'll take your comments on board when we next review this component.

**Comment by rinto-c** [comment]
I am trialling a few different options for the 'Menu' as users did not recognise/found the menu on the header. Bringing the menu below the header also has a caveat of phase banner dominating the place and hence it would be easy for users to miss the menu.

I am quite keen to understand if we can remove the phase banner from below the header and move it just above the footer.

![image](https://user-images.githubusercontent.com/64977675/109977559-d7d72e00-7cf4-11eb-8e07-a69583bd9773.png)

**Comment by lukechaput** [comment]
On a previous service our users (and indeed a large number of our internal stakeholders) did not understand what things like 'alpha' and 'beta' meant. Very few of our users going into private beta even noticed the banner in testing. 

We opted to:

- frontload the banner content with the thing we wanted users to do ('give feedback')
- change the label colour so it was more noticeable
- remove the word 'beta' from the label

<img width="457" alt="Screenshot 2021-05-20 at 13 42 43" src="https://user-images.githubusercontent.com/38037503/118980452-64e36800-b971-11eb-9f25-7e7581b09768.png">

This tested better in the lab (...although it was unclear whether it actually encouraged any more users to give feedback.)

Minor other point: not sure why there's guidance for an alpha banner when this would only ever been seen in the lab. Adding it to the design system makes it seem like there's instances when we'd want to solicit survey feedback from users in alpha - and I can't think of a situation where we would.

**Comment by MalcolmVonMoJ** [comment]
There are some services that are "live" but still claim to be in Alpha.  

e.g. [DartCharge](https://www.dartford-crossing-charge.service.gov.uk/Security/SignIn)

**Comment by lukechaput** [comment]
I'm not really sure how that works. Certainly runs counter to what we're trying to tell people about phasing their service, meeting the Standards, passing assessments, etc. If it is an unfortunate anomaly, should it be part of the x-gov pattern?

**Comment by ghost** [comment]
We're currently testing the phase banner with the hypothesis that 'users that are not in the digital industry don’t know what the terms “alpha” or “beta” mean. Instead, users would understand the term “new service” and that could lead to higher engagement rates on the feedback link and reduce confusion.'

**Questions we're asking:**
- Do users understand the term “alpha” and “beta”? 
- Have users seen that banner before? If so on what services? 
- Which banner do they prefer? 

**Results so far:**
- 3 out of 4 users have seen the banner before
- All 4 users don't understand the terms beta and alpha terms
- 1 user doesn't mind which option, 2 prefer the new option, 1 prefered the old option (even though they didn't understand the term)

![before-after](https://user-images.githubusercontent.com/50828707/119311434-f9e3ab00-bc68-11eb-8293-bca54c88d500.png)

**Comment by terrysimpson99** [comment]
Live services don't get a banner at all. Which is odd. Perhaps we should use a banner that says "MATURE SERVICE. We don't want any feedback". Half joking.

**Comment by matthewford** [comment]
Hi, just to bring up the placement of the phase banner again, along similar lines to @rinto-c 's question, is the intention of the phase banner to be placed after the main navigation?

**Comment by murfious** [comment]
The Get a fishing licence service has passed live assessment but our team do not want to lose the feedback link. We think the feedback is useful for users and for the service. Our current plan is to remove the status badge and keep the feedback text, as shown below.

<img width="1022" alt="Screenshot 2021-07-27 at 16 15 29" src="https://user-images.githubusercontent.com/9218852/127180374-ab3d0807-170a-4f8a-a6b7-89bc9994602e.png">

**Comment by frankieroberto** [comment]
@murfious congrats on passing the live assessment! Out of curiosity, can you give a rough indication of how many users (number or percentage) give feedback via the link? I've always found it to be quite low...!

**Comment by murfious** [comment]
@frankieroberto, from May 2020 to April 2021 we had just over 1.5M users.
In the same period we had just over 10K (10,345) counts of feedback, so that seems to be 0.66%.

Just under half of the instances of feedback had written comments.

Some people may visit twice to buy multiple licences but we think those numbers are low, although we can not yet tell. We are currently working on something to enable users to buy several licences in one pass.

**Comment by paulmsmith** [comment]
Adding some notes that might be helpful to others.

The service I'm working on, is currently in Alpha but has a 'live' (publicly accessible) aspect to it. The reason for this is because our service is also a 'platform' (a service to help people build other services) which necessitates it being publicly accessible to the users we're designing for and researching with, so that our team can test, try and throw-away various incarnations of it.

Currently I'm testing a hypothesis that the content of our phase banner needs to emphasise the rapidly changing functionality as well as content of the platform/service. In other words, somehow tell users that it is a risk to form a critical dependancy on it (in whatever form it currently takes for them) during Alpha/Beta phases. 

Will post back when we have more.

**Comment by wilsond-gds** [comment]
Hopefully this is the best place to report this rather than the colours discussion?

We’ve had an accessibility audit on the beta version of the digital identity service and the audit picked up that the colour contrast is 5.16:1 between feedback link (#1d70b8) and background (#fff). This passes AA but fails AAA.

Please let me know if you’d like to see the full section of the report for more information.

**Comment by stevenjmesser** [comment]
Personal opinion but I think the feedback link should be removed from the phase banner. There are better moments in a journey to collect feedback, usually at the end of a transaction or set of tasks, so that we don't distract the user from achieving their outcome.

Stephen Gill [wrote about it](https://medium.com/@stephen.gill/give-help-now-get-feedback-later-af423fff3fb1) recently.

**Comment by stevenjmesser** [comment]
It'd be great to collect more [evidence on how users don't understand the phase banner](https://github.com/alphagov/govuk-design-system/issues/4972#issuecomment-3551731935), especially the alpha and beta terminology. It's currently a poorly evidenced hypothesis!

**Comment by OscarJacksonCAP** [comment]
Just wondering if anything ever came of the discussion about moving the link to the end of the line.
I've seen it raised further up this discussion so just wondering if it was investigated and, if so, what was the outcome?

**Comment by andymarshall-35** [comment]
Has the placement of the Phase banner been reviewed in relation to other items that might sit in the same spot, specifically the Primary navigation component that I understand is making its way through to x-gov?
I'd been working on a page that included both, and the current guidance seems to suggest the phase banner sits between the header and primary nav [1.], but given that breaks the hierarchy of the header > nav > content, I felt more inclined to move the phase banner either above the header (contraversial) [2.], or below the primary nav (preferable) [3.] - see attached:
1.
<img width="1033" alt="Screenshot 2024-05-15 at 15 26 52" src="https://github.com/alphagov/govuk-design-system/assets/167748704/53147fc0-2f29-44e5-ad39-179086790937">

2.
<img width="1041" alt="Screenshot 2024-05-15 at 15 27 11" src="https://github.com/alphagov/govuk-design-system/assets/167748704/39dbd866-ddf4-4669-ab09-01b82cc1bd49">

3.
<img width="1080" alt="Screenshot 2024-05-15 at 16 24 24" src="https://github.com/alphagov/govuk-design-system/assets/167748704/b78d23d0-10a7-49dc-afa7-2b54d025b2fe">

**Comment by vickytnz** [comment]
2 things:

1. I've proposed moving the phase banner to the footer and changing it here:
https://github.com/alphagov/govuk-design-system-backlog/issues/96#issuecomment-2266695170
2. I think there's ambiguous language about the phase banner and services that are assessed to the service standard but aren't on the govuk domain - by govuk design system standards they don't need a phase banner, but by the service standard they are beta not live and should be iterating with a full-time team, so it comes down to the intent of the banner

**Comment by markjefferson-gov** [comment]
Does anyone have any insights on the effectiveness of the phase banner for generating feedback?

I have a hypothesis that a lot of services use the phase banner as a convenient place to stick a feedback link rather than from any need to indicate the development status of the service (which I think we all agree doesn't mean much to users).

A key benefit seems to be that it enables you to provide a feedback link on every page but I wonder:

* is it effective at soliciting feedback? My limited experience suggests not very
* is a feedback link needed on every page? I'd imagine we generally don't want to interrupt users during their tasks

Could a specific feedback pattern or component targeted at the right place in a user journey result in more and better feedback?

**Comment by scottywakefield** [comment]
I recommend adding attr role="status" to avoid accessibility violations. I have had a violation reported of "Ensure all page content is contained by landmarks".

**Comment by querkmachine** [comment]
@scottywakefield We have an issue where that was raised and discussed here: https://github.com/alphagov/govuk-frontend/issues/1604. 

The short version, however, is that the "violation" is really just a case of Axe being overzealous in how it reports things: it is not a failure of WCAG for content to be outside of a landmark and it makes very little difference to an assistive technology user whether the phase banner is in a landmark or not either, as it's not content that a user will typically be trying to navigate to.

Regardless, the `status` role is not appropriate to use on the phase banner. That role is intended for status messaging—elements that provide live, dynamically updating content about the current state of the page—and includes implicit behaviours for that functionality that could affect how the phase banner gets announced to users. 

Assuming Axe is what flagged this to you, more recent versions of it should no longer flag this as a violation, but as a best practice instead. If you still want to have _some_ role on the element, then the `complementary` role is probably the most suitable, though that also isn't a perfect fit.

**Comment by scottywakefield** [comment]
@querkmachine Thanks for your detailed answer!! Yes we are using axe. I have gone with wrapping the phase banner (and anything else we want to skip over with the skip link) in an `aside` element for now.

**Comment by johnwatson484** [comment]
> @querkmachine Thanks for your detailed answer!! Yes we are using axe. I have gone with wrapping the phase banner (and anything else we want to skip over with the skip link) in an aside element for now.

@scottywakefield  

We had the same solution.

Overzealous Axe or not, adding clearer content intent through the `aside` landmark isn't a bad thing.

**Comment by LydiaTeebay** [comment]
Has anyone thought of using this phase banner as a way to allow users to know that AI is driving the work? In CPS I'm trying to add in something that I think meets the same needs for adding phase banners. Thoughts?
<img width="1133" height="198" alt="Screenshot 2026-01-28 at 14 38 53" src="https://github.com/user-attachments/assets/aa6a641f-aad0-40dd-8c62-06341a778cd4" />

**Comment by sarahfisherg** [comment]
The example given for “give your feedback by email” - with that text being the link text, contradicts the GDS style guide:
"email addresses
Write email addresses in full, in lower case and as active links. Do not include any other words in the link text”

Hiding the email address behind nice user friendly text seems like a nice design, but is problematic and bad usability because:

- it means people can't cut and paste the email if they don't want it to fire up an email client automatically
- users don't know what it will do next
- it's what scammers do

The example should instead be something like:
 ...give your feedback by email to [example.email@gov.uk](example.email@gov.uk)
